### PR TITLE
refactor: simplify git version check

### LINKED ISSUE #17695: Simplify git version check
### Describe the proposed change(s).

Git version check can be simplified with `simple-git@3.14.0`[^1].
Update is currenly pending.

[^1]: https://github.com/steveukx/git-js/blob/main/examples/git-version.md

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Context
- Closes : #17695 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — CHANGES_REQUESTED** [review]
there is another one in lib/utils/git, which needs to be updated

**Comment by viceice** [comment]
https://github.com/renovatebot/renovate/blob/c4fa239bfaf61deef30c7d2f95afd77f5d4a7acf/lib/util/git/index.ts#L164-L195

File: lib/util/git/index.ts [hunk 1]
```diff
@@ -165,13 +165,8 @@ export async function validateGitVersion(): Promise<boolean> {
...
   try {
-    const raw = await globalGit.raw(['--version']);
-    for (const section of raw.split(/\s+/)) {
-      if (semverCoerced.isVersion(section)) {
-        version = section;
-        break;
-      }
-    }
+    const { major, minor, patch } = await globalGit.version();
+    version = `${String(major)}.${String(minor)}.${String(patch)}`;
```
Comment (viceice) [inline lib/util/git/index.ts:169]:
```suggestion
    version = `${major}.${minor}.${patch}`;
```

why those string stuff?

Comment (RahulGautamSingh) [reply lib/util/git/index.ts:169]: TS string literal error.

Comment (RahulGautamSingh) [reply lib/util/git/index.ts:169]: Inferred type for major, minor and patch are bumber only.Issue isn't for null. 
It is that inside template literal expression we cannot use variable which is other than string type.

Comment (viceice) [reply lib/util/git/index.ts:169]: number should be ok. 🤔

Comment (RahulGautamSingh) [reply lib/util/git/index.ts]: It doesn't throw the error now🙂. Sometimes it does and other times it doesn't ...weird.

File: tools/check-git-version.mjs [hunk 2]
```diff
@@ -7,10 +7,9 @@ const git = simpleGit();
 
 await (async () => {
   try {
-    const regex = /\d+\.\d+\.\d+/;
-    const stdout = await git.raw('--version');
-    const [gitVersion] = regex.exec(stdout) ?? [];
-    if (!gitVersion || semver.lt(gitVersion, GIT_MINIMUM_VERSION)) {
+    const { major, minor, patch, installed } = await git.version();
+    const gitVersion = `${major}.${minor}.${patch}`;
+    if (!installed || semver.lt(String(gitVersion), GIT_MINIMUM_VERSION)) {
```
Comment (viceice) [inline tools/check-git-version.mjs:12]:
```suggestion
    if (!installed || semver.lt(gitVersion, GIT_MINIMUM_VERSION)) {
```
it's always a string

**Review by viceice — APPROVED** [review]

**Pull request merged**